MapboxCoreNavigation/RouteController.swift

```diff
@@ -541,6 +543,28 @@ extension RouteController: CLLocationManagerDelegate {
         let radius = max(reroutingTolerance, location.horizontalAccuracy + RouteControllerUserLocationSnappingDistance)
         let isCloseToCurrentStep = newLocation.isWithin(radius, of: routeProgress.currentLegProgress.currentStep)
         
+        
+        // Check to see if the user is moving away from the maneuver.
+        // Here, we store an array of distances. If the current distance is greater than the last distance,
+        // add it to the array. If the array grows larger than x, reroute the user.
+        if let coordinates = routeProgress.currentLegProgress.currentStep.coordinates {
+            let userDistanceToManeuver = distance(along: coordinates, from: location.coordinate)
+            
+            guard previousDistancesFromManeuver.count <= 3 else {
+                resetPreviousDistanceArray()
```

Comment:
We should make it so that rerouting automatically calls this method, making this line redundant.